datafusion/optimizer/src/optimizer.rs

```diff
@@ -44,15 +44,23 @@ pub struct OptimizerConfig {
     /// Query execution start time that can be used to rewrite expressions such as `now()`
     /// to use a literal value instead
     pub query_execution_start_time: DateTime<Utc>,
+    next_id: usize,
 }
 
 impl OptimizerConfig {
     /// Create optimizer config
     pub fn new() -> Self {
         Self {
             query_execution_start_time: chrono::Utc::now(),
+            next_id: 1,
```

Comment:
This is used for creating a unique alias nr. right? Maybe we can add a comment that it serves this purpose.